spec/abci++/abci++_basic_concepts_002_draft.md

```diff
@@ -4,32 +4,51 @@ title: Overview and basic concepts
 ---
 
 ## Outline
+
 - [ABCI++ vs. ABCI](#abci-vs-abci)
-- [Methods overview](#methods-overview)
-  - [Consensus methods](#consensus-methods)
-  - [Mempool methods](#mempool-methods)
-  - [Info methods](#info-methods)
-  - [State-sync methods](#state-sync-methods)
+- [Method overview](#method-overview)
+    - [Consensus/block execution methods](#consensusblock-execution-methods)
+    - [Mempool methods](#mempool-methods)
+    - [Info methods](#info-methods)
+    - [State-sync methods](#state-sync-methods)
 - [Next-block execution vs. same-block execution](#next-block-execution-vs-same-block-execution)
-  - [Tendermint timeouts](#tendermint-timeouts-in-same-block-execution)
-- [Determinism](#determinism)
-- [Errors](#errors)
+- [Tendermint proposal timeout](#tendermint-proposal-timeout)
+- [Deterministic State-Machine Replication](#deterministic-state-machine-replication)
 - [Events](#events)
 - [Evidence](#evidence)
+- [Errors](#errors)
 
 # Overview and basic concepts
 
 ## ABCI++ vs. ABCI
-[&uparrow; Back to Outline](#outline)
 
-With ABCI, the application can only act at one phase in consensus, immediately after a block has been finalized. This restriction on the application prevents numerous features for the application, including many scalability improvements that are now better understood than when ABCI was first written. For example, many of the scalability proposals can be boiled down to "Make the miner / block proposers / validators do work, so the network does not have to". This includes optimizations such as tx-level signature aggregation, state transition proofs, etc. Furthermore, many new security properties cannot be achieved in the current paradigm, as the application cannot enforce validators to do more than just finalize txs. This includes features such as threshold cryptography, and guaranteed IBC connection attempts.
+[&uparrow; Back to Outline](#outline)
 
-ABCI++ overcomes these limitations by allowing the application to intervene at three key places of the block execution. The new interface allows block proposers to perform application-dependent work in a block through the `PrepareProposal` method; validators to perform application-dependent work in a proposed block through the `ProcessProposal` method; and applications to require their validators do more than just validate blocks, e.g., validator guaranteed IBC connection attempts, through the `ExtendVote` and `VerifyVoteExtension` methods. Furthermore, ABCI++ renames {`BeginBlock`, [`DeliverTx`], `EndBlock`} to `FinalizeBlock`, as a simplified way to deliver a decided block to the Application.
+With ABCI, the application can only act at one phase in consensus, immediately after a block has
```

Comment:
IMO, legacy ABCI vs ABCI is a nicer distinction. Once ABCI++ settles, users will not know about the old thing and referencing it forever seems a little silly.

Reply:
The decision we reached in one of the ABCI++ calls in January was this:
* for v0.36 we would keep the distinction ABCI vs ABCI++ to avoid any confusion (as both are actually distinct interfaces albeit with some common elements).
* then, once v0.36 was out for a while, and the time came to prepare for v0.37 we would decide whether ABCI (pre-v0.36) was really never to be seen again, at which point we might decide to rename ABCI++ into ABCI in our docs from v0.37 on.